Aflores plymouth (#234)
osisoft/Edge-Data-Store-Docs: content/egress/egress-execution-details.md

# Egress execution details

> After an egress endpoint is configured, data egress occurs periodically for that endpoint based on its configuration and independently from other endpoints.

Reviewer: ```suggestion
After you configure an egress endpoint, data egress occurs periodically and independently from other endpoints based on its configuration .
```

> **Note:** Only streams with a single, timeseries-based index can be egressed.

Reviewer: ```suggestion
**Note:** EDS can only egress streams with a single timeseries-based index.
```

> EDS uses OMF messages to egress data, and knowing how those messages are constructed can help with understanding how data is egressed. OMF defines three types of messages: types, containers, and data. Types and containers define the data being egressed and data is the actual timeseries data. For EDS, types and containers are sent only on the first egress for an endpoint; subsequently, only new or changed types and containers are egressed. Type creation must be successful to perform container creation, and container creation must be successful to perform data egress.

Reviewer: ```suggestion
EDS uses OMF messages to egress data and knowing how those messages are constructed can help with understanding how data is egressed. OMF defines three types of messages: types, containers, and data. Types and containers define the egressed data and data is the actual timeseries data. For EDS, types and containers are sent only on the first egress for an endpoint; subsequently, only new or changed types and containers are egressed. Type creation must be successful to perform container creation, and container creation must be successful to perform data egress.
```

- HttpStatusCode GatewayTimeout (504)

> There is an increasing delay before each retry. The total time for all five retries is one minute; during which time, all other egress messages for this definition are queued. If a TimeoutException or HttpRequestException error still occurs at the fifth retry, the egress action fails, and EDS waits five minutes before trying the whole process again. If one of the other errors still occurs at the fifth retry, the message fails to send and EDS will move on to the next message in the queue.

Reviewer: ```suggestion
There is an increasing delay before each retry. The total time for all five retries is one minute. During this time, EDS queues all other egress messages for this definition. If a `TimeoutException` or `HttpRequestException` error still occurs at the fifth retry, the egress action fails and EDS waits five minutes before trying the whole process again. If one of the other errors still occurs at the fifth retry, the message fails to send and EDS moves on to the next message in the queue.
```

> For data collection and egress, in-memory and on-disk storage are used to track the last successfully-egressed data event, per stream. Data is egressed in the order it is collected, and egress configurations can include future data.

Reviewer: ```suggestion
For data collection and egress, in-memory and on-disk storage are used to track the last successfully-egressed data event per stream. EDS egresses data in the order it is collected and egress configurations can include future data.
```

> **Note:** When an event with a future timestamp is successfully egressed, only values after the associated timestamp of that event will be egressed.

Reviewer: ```suggestion
**Note:** When EDS successfully egresses an event with a future timestamp, only values after the associated timestamp of that event are egressed.
```
